client/overview/task-list/dispute-tasks.ts

```diff
@@ -32,7 +32,10 @@ const isDueWithin = ( dispute: CachedDispute, days: number ) => {
 	// Get current time in UTC.
 	const now = moment().utc();
 	const dueBy = moment.utc( dispute.due_by );
-	return dueBy.diff( now, 'hours' ) > 0 && dueBy.diff( now, 'days' ) <= days;
+	return (
+		dueBy.diff( now, 'hours', true ) > 0 &&
+		dueBy.diff( now, 'days', true ) <= days
+	);
```

Comment:
This fixes a minor bug in the dispute task `isDueWithin()` logic – it will now correctly return `false` if it is not _within_ the desired number of days/hours.

> Previously, it would return `true` if asked for disputes within 3 days and a dispute was due between 3-4 days / 72-96 hours.
> In this PR, it will return `false` in this case, since a dispute due in 3-4 days is not due within 3 days.


See [here](https://momentjs.com/docs/#/displaying/difference/) for usage of the third `moment.diff` `precise` arg.

Reply:
Yeh I'm finding the logic for these date comparisons to be continuously difficult to reason with.

The `isDueWithin` util function + unit tests cleared this up for me – [another +1 to roll that out across disputes codebase](https://github.com/orgs/Automattic/projects/578/views/6?pane=issue&itemId=33155814).